packages/chain-adapters/src/utxo/UtxoBaseAdapter.ts

```diff
@@ -620,4 +608,18 @@ export abstract class UtxoBaseAdapter<T extends UtxoChainId> implements IChainAd
 
     return publicKeys[0]
   }
+
+  async verifySignTx(tx: SignTx<T>) {
+    const addresses = (
+      await Promise.all(
+        tx.inputs.map(async input => {
+          const tx = await this.providers.http.getTransaction(input)
+          // assumes input.vout is an index to tx.vout[]
+          return tx.vout[input.vout].addresses ?? []
+        }),
+      )
+    ).flat()
```

Comment:
would appreciate input on this - is it correct?

Reply:
We will want to verify both inputs and outputs UTXOs here @woodenfurniture meaning no UTXO is consumed with a sanctioned address, as well as none of the outputs being to a sanctioned address.

tl;dr of how things work is we're augmenting `coinselect` in our `utxoSelect` implementation, and then again this augmented version is augmented in `signTxInputs` / `signTxOutputs`, bringing in additonal metadata.

Ideally, `utxoSelect` itself would be the one doing those checks so we don't have to deep-introspect the `SignTx` type here, though that would get bloaty.
Preferrably, if we don't want to bloat `utxoSelect` with this, we can do the verification logic on the destructured `inputs` and `outputs` of `coinSelectResult`. This will give us direct access to `input.address` for each input.